### PR TITLE
Changing CSharpCodeGeneratorHelper to be a service which is passed in.

### PR DESCRIPTION
**Pull request by lajones**
See issue #830. Updating CSharpCodeGeneratorHelper to no longer be a singleton and instead be a service which is passed in.

### CODE REVIEW

File: src/EntityFramework.Relational.Design/ReverseEngineering/ReverseEngineeringGenerator.cs [hunk 1]
```diff
@@ -22,15 +23,24 @@ public class ReverseEngineeringGenerator
...
             if (Logger == null)
             {
                 throw new ArgumentException(typeof(ReverseEngineeringGenerator).Name + " cannot find a service of type " + typeof(ILogger).Name);
             }
+
+            CSharpCodeGeneratorHelper = (CSharpCodeGeneratorHelper)serviceProvider.GetService(typeof(CSharpCodeGeneratorHelper));
+            if (CSharpCodeGeneratorHelper == null)
+            {
+                throw new ArgumentException(typeof(ReverseEngineeringGenerator).Name + " cannot find a service of type " + typeof(CSharpCodeGeneratorHelper).Name);
+            }
```
Comment (bricelam) [inline src/EntityFramework.Relational.Design/ReverseEngineering/ReverseEngineeringGenerator.cs]: There's an extension method for doing this--`GetRequiredService<T>()`.

**Comment by bricelam** [comment]
:shipit:

File: src/EntityFramework.Relational.Design/ReverseEngineering/ReverseEngineeringGenerator.cs (hunk 1 above)
Comment (lajones) [inline src/EntityFramework.Relational.Design/ReverseEngineering/ReverseEngineeringGenerator.cs]: Will do. Thanks.

**Comment by lajones** [comment]
Updated and checked in - commit d5f5629.

**Pull request closed without merging**